microsoftgraph/client.py

```diff
@@ -221,10 +221,7 @@ def _request(self, method, url, headers=None, **kwargs):
             _headers.update(headers)
         if self.requests_hooks:
             kwargs.update({"hooks": self.requests_hooks})
-        if "files" not in kwargs:
```

Comment:
Rather than removing this block completely, I would update this check to: 
`if "Content-Type" not in _headers:`

So in case we get already a content type provided in the headers we preserve it, but in case no content-type is specified, we apply application/json, which is the default one for this rest api.